### PR TITLE
include more versions, little regex fixes, improve OS detection, arch…

### PR DESCRIPTION
**Pull request by m-1-k-3**
* The usual version detection updates
* Extend pre checker for better OS detection

image

* Extend pre checker for architecture detection on RTOS:
 
image

* These details are also used in the final aggregator:

image

* config/bin_version_strings.cfg now supports the marker **strict** and **binary**. "binary" is for version identifiers that are used on binary files (in the modules modules/P09_firmware_base_version_check.sh and modules/S09_firmware_base_version_check.sh). This means all version identifiers are used in the emulation module. In the binary modules only identifiers with "binary" are used.

### CODE REVIEW

File: modules/P07_firmware_bin_base_analyzer.sh [hunk 1]
```diff
@@ -15,48 +15,108 @@
...
+  if ! [[ -d "$FIRMWARE_PATH" ]]; then
+    binary_architecture_detection
+  fi
 }
 
 os_identification() {
+  sub_module_title "OS detection"
+
+  print_output "[*] Initial OS detection running " | tr -d "\n"
+  print_output "." | tr -d "\n"
```
Comment (p4cx) [inline modules/P07_firmware_bin_base_analyzer.sh:32]: Suggestion: Write a small function for `print_output "." | tr -d "\n"` in `print.sh`. These dots also don't have to be written into log. Out Log looks quite stupid with these dots :D 
Excerpt:
```
[[0;33m*[0m] Initial OS detection running 
.
.
.
.
.
.
.
.
```

**Review by p4cx — CHANGES_REQUESTED** [review]

File: modules/P07_firmware_bin_base_analyzer.sh (hunk 1 above)
Comment (p4cx) [reply modules/P07_firmware_bin_base_analyzer.sh:32]: Also please replace `print_output "." | tr -d "\n"` with the new function call in the `P09` module. Thanks.

Comment (m-1-k-3) [reply modules/P07_firmware_bin_base_analyzer.sh:32]: woo .. thats really ugly. Let's switch to an "echo" for the dots

**Comment by m-1-k-3** [comment]
the logs are clean again ...

**Review by p4cx — APPROVED** [review]

**Pull request merged**